trace-cruncher: Introduce Docker installation

knauth opened the pull request:
This PR continues work towards making trace-cruncher fully reproducible and containerized.

The Dockerfile builds a Debian image with trace-cruncher installed, using the new date-snapshot script to freeze the built dependencies as well as trace-cruncher itself.

The next phase of this work will be moving to a Debuerreotype base image so that the container build will be completely reproducible.

Additionally this commit also updates the README with basic Docker install instructions.

Still to-do is to remove build dependencies after the build is complete. Also, the script will need to be updated once the date-snapshot script is changed to use commit tags. It might be worthwhile to create a "repos-docker" file which includes tracecruncher and have the script fetch from that file instead of calling the script twice. For those reasons I am marking this as a draft.

However, I wanted to put in this PR before the long weekend to get feedback on the Dockerfile architecture. As opposed to using the date-snapshot script, it would be possible to lock the version manually in the Dockerfile and `git clone` the repositories directly in the file itself. However, this would mean another place where we need to update our dependencies. I like the idea of having one file to update that we know affects all our builds, both manual and Dockerfile. We might explore environment variables via GitHub itself, which could provide a centralized location for locking versions. Please discuss.

Signed-off-by: June Knauth (VMware) <june.knauth@gmail.com>

knauth committed:
trace-cruncher: Introduce Docker installation

This commit continues work towards making trace-cruncher fully reproducible and containerized.

The Dockerfile builds a Debian image with trace-cruncher installed, using the new date-snapshot script to freeze the built dependencies as well as trace-cruncher itself.

The next phase of this work will be moving to a Debuerreotype base image so that the container build will be completely reproducible.

Additionally this commit also updates the README with basic Docker install instructions.

Signed-off-by: June Knauth (VMware) <june.knauth@gmail.com>

warthog9 commented on Dockerfile:
```diff
@@ -0,0 +1,22 @@
+# SPDX-License-Identifier: LGPL-2.1
+# Copyright (C) 2022, VMware Inc, June Knauth <june.knauth@gmail.com>
+
+FROM debian
```
We should probably pin a major version at least

warthog9 reviewed (changes requested):
I think just locking Debian to a version should do it for now

knauth committed:
trace-cruncher: Tag Dockerfile debian version and remove build deps

This commit removes the build dependencies at the end of the Dockerfile, reducing
image size by about 400 MB.

Additionally it locks the Debian version to bullseye as an intermediate step towards Debuerreotype integration.

Signed-off-by: June Knauth (VMware) <june.knauth@gmail.com>
